samples/seat_massager/provider/src/request_impl.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+use digital_twin_model::sdv_v2 as sdv;
+use log::{debug, info, warn};
+use parking_lot::Mutex;
+use samples_protobuf_data_access::async_rpc::v1::request::{
+    request_server::Request, AskRequest, AskResponse, NotifyRequest, NotifyResponse,
+};
+use samples_protobuf_data_access::async_rpc::v1::respond::{
+    respond_client::RespondClient, AnswerRequest,
+};
+use seat_massager_common::TargetedPayload;
+use std::sync::Arc;
+
+#[derive(Debug, Default)]
+pub struct RequestState {}
+
+#[derive(Debug, Default)]
+pub struct RequestImpl {
+    pub state: Arc<Mutex<RequestState>>,
+}
+
+impl RequestImpl {}
+
+#[tonic::async_trait]
+impl Request for RequestImpl {
+    /// Ask implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Ask request.
+    async fn ask(
+        &self,
+        request: tonic::Request<AskRequest>,
+    ) -> Result<tonic::Response<AskResponse>, tonic::Status> {
+        let request_inner = request.into_inner();
+        let respond_uri: String = request_inner.respond_uri.clone();
+        let ask_id: String = request_inner.ask_id.clone();
+        let payload: String = request_inner.payload.clone();
+
+        info!("Received an ask request");
+        info!("respond_uri: {respond_uri}");
+        info!("ask_id: {ask_id}");
+        info!("payload: {payload}");
+
+        let targetted_payload_json: TargetedPayload = serde_json::from_str(&payload).unwrap();
+        info!("instance_id: {}", targetted_payload_json.instance_id);
+        info!("member_path: {}", targetted_payload_json.member_path);
+        info!("operation: {}", targetted_payload_json.operation);
+        info!("inner payload: {}", targetted_payload_json.payload);
```

Comment:
Similar to the above, if `TargetedPayload` implements `Debug` you can do this:

```rust
info!("Payload: {targeted_payload:#?}");
```

Reply:
This does not cleanly render the inner payload value.  Leaving it as is.

Reply:
Can you provide more detail? The pretty-print operator `#` in the format string will apply formatting and print each property of `targeted_payload` on a new line with indentation, making it easily readable

Reply:
I no longer want to print the payload value.